src/markdown_generators_test.js

```diff
@@ -267,6 +278,21 @@ describe("markdown generators", () => {
         }
       })
     })
+
+    it("resolves urls inside embedded media urls", () => {
+      const markdownData = markdownGenerators.generateMarkdownFromJson(
+        subtitlesCourseJsonData,
+        pathLookup
+      )
+
+      const embeddedMedia = yaml.safeLoad(
+        markdownData[3].media[0].data.split("---\n")[1]
+      )
+      assert.equal(
+        embeddedMedia.embedded_media[6].technical_location,
+        "BASEURL_SHORTCODE/sections/instructor-insights/instructor-interview-meet-the-educator/m_gqolc3clm"
```

Comment:
The `baseurl` shortcode is only for use directly within markdown content as `{{< baseurl >}}`.  Since these links are going in front matter, the shortcode won't be applied.  Instead, templates that use this data need to prepend where they are writing the link with `.Site.BaseURL` if it's a relative link.

Reply:
Hmm, so that means the link should start with `/sections/...`? How would the template know whether to prepend `.Site.BaseURL` or leave it as is, for example with `/ans7870` urls?

Reply:
What do you think about having ocw-to-hugo ask for a baseurl and prepend it there instead of in hugo?

Reply:
I'm not sure how I feel about that, as the content loses its "portability" at that point.  On the production server (right now, at least) we're building fresh content using `ocw-to-hugo` every time.  If we were asking for a `baseUrl` argument in `ocw-to-hugo` and writing that into some links, that markdown wouldn't be usable in build contexts other than the specific context you passed in that baseUrl for.  It's maybe not such a big deal. but it strikes me as going against the grain a bit in the way static site publishing is supposed to work. Either way, I think it's maybe a topic for another PR.

Reply:
Another option might be rendering the links with `{{< baseurl >}}` in the front, and then when you iterate them in the template you could do something like:

```
{{ range .Params.embedded_media }}
  <a href="{{ .technical_location | markdownify }}">{{ .id }}</a>
{{ end }}
```

This would put the link text through the markdown preprocessor and give you the base URL.

Reply:
@noisecapella looks like some of those elements have the url stored on `media_location` so you'd have to check that as well, but I think putting the literal shortcode into the url and piping into `markdownify` in the Hugo template should make that work.  Want to give it a shot?

Reply:
This is also worth looking into: https://gohugo.io/functions/renderstring/

Reply:
Disregard what I said above about `markdownify`, I just tried it and it doesn't work in this context.  It seems that shortcodes are really only processed on the `.Content` property of markdown files.

Reply:
So I was digging more into this particular example, and it seems that `m_gqolc3clm` is the `media_location` property on an `embedded_media` element of type `Video-YouTube-Stream`.  In this case, I don't think the URL needs to be rewritten, as it's not really a URL at all.  This is just used to fetch a video from YouTube by its ID.  I don't think we need to be rewriting urls that don't start with `https://ocw.mit.edu`.

Reply:
The assertion here is actually looking at the URL for `acf8cbbcf3ada9e7b0d390c8b4f8b1e6`, which is originally `https://ocw.mit.edu/courses/global-languages/21g-107-chinese-i-streamlined-fall-2014/instructor-insights/instructor-interview-meet-the-educator/M_gQolc3clM.pdf`.  Shouldn't this be re-written to an S3 URL pointing to the PDF?

Reply:
The S3 link to this PDF is available in the course's parsed JSON:

```
        {
            "order_index": null,
            "uid": "acf8cbbcf3ada9e7b0d390c8b4f8b1e6",
            "id": "M_gQolc3clM.pdf",
            "parent_uid": "58a6c062c3801594c1010c9bacadde98",
            "title": "3play pdf file",
            "caption": null,
            "file_type": "application/pdf",
            "alt_text": null,
            "credit": null,
            "platform_requirements": "",
            "description": "",
            "type": "OCWFile",
            "file_location": "https://open-learning-course-data-production.s3.amazonaws.com/21g-107-chinese-i-streamlined-fall-2014/acf8cbbcf3ada9e7b0d390c8b4f8b1e6_M_gQolc3clM.pdf"
        },
```

Reply:
@gumaerc This should be fixed. I changed it to always use a S3 link for PDFs. Regarding `media_location`, I did a grep and couldn't find any URLs for `media_location` which start with `http://ocw.mit.edu` or `https://ocw.mit.edu`